packages/prettier-plugin-java/src/printers/prettier-builder.js

```diff
@@ -77,31 +77,44 @@ function processComments(docs) {
 }
 
 function concat(docs) {
-  return prettier.concat(processComments(docs));
+  const concatenation = prettier.concat(processComments(docs));
+  return concatenation.parts.length === 0 ? "" : concatenation;
```

Comment:
IMO we should return ```undefined``` instead of ```""``` since we return an object and not a string.
It is also easier to use as we won't have to check with ```if(concat(docs) === "")``` but directory ```if(concat(docs))```

Reply:
I use the same "convention" as here :

https://github.com/clement26695/prettier-java/blob/35c11f4656af8ba1e591e52e25a6b0f42d95a683/packages/prettier-plugin-java/src/cst-printer.js#L69-L74

And IMO, it is the same for your second argument (testing if (elt) will check if elt === undefined/""/[] etc.)

But you're rigth, it migth be clearer...